50. Pow(x, n)/Solution.java

```diff
@@ -0,0 +1,37 @@
+class Solution {
+    
+    public double myPow(double x, int n) {
+        // call helper method
+        return myPow(x, (long) n);
+    }
+    
+    // helper method
+    private double myPow(double x, long n) {
+        // case: n = 0
+        if(n == 0) {
+            return 1;
+        }
+        // case: n = 1
+        if(n == 1) {
+            return x;
+        }
+        // case: n < 0, find nth power of reciprocal of x
+        // changing type of n to long takes care of cases like when n == Integer.MIN_VALUE
+        // otherwise when sign flips we run into overflow 
+        if(n < 0) {
+            x = 1/x;
```

Comment:
This is equivalent to 
 ` if(n<0){
            return 1/pow(x,-n);
        }`